### PR TITLE
Approutes

### PR DESCRIPTION
**Pull request by gzinck**
This lays the foundations for the routes; the front-end can now ping the server with requests at various URLs to automatically query the database and get a response. Routes are presently only set up for getting user information and dealing with trips, with more to come. The current setup means that all of the legwork is done in a couple very generic functions in databaseHandler, and other more specific handlers are essentially wrappers that pass custom queries to the databaseHandler.

### CODE REVIEW

**Comment by evscott** [comment]
One more comment, it looks like compiled files (dist) snuck into this PR somehow. I assume it has to do with the directory not being in our .gitignore. I can retroactively fix this after you merge.

File: server/handlers/tripHandler.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Updates a pre-existing trip from the database.
+ * @param req the request
+ * @param res the resource to give the response
+ * @returns {Promise<void>} the promise indicating success
+ */
+let updateTrip = async (req, res) => {
+  const username = pool.escape(req.body.username); // Get username from body
+  const requestedTrip = req.params['tname'];
+  console.log(requestedTrip); // TODO: remove this
```
Comment (evscott) [inline server/handlers/tripHandler.js]: Should this still be here?

**Comment by gzinck** [comment]
Excellent! And thanks, adding the gitignore after would be good.

File: server/handlers/tripHandler.js (hunk 1 above)
Comment (gzinck) [reply server/handlers/tripHandler.js]: Good point! The TODO to remove the console.log doesn't really need to be removed since it's on the server.

**Comment by evscott** [comment]
Awesome. Merge away!

**Pull request merged**